scripts/docker-entrypoint.sh

```diff
@@ -45,6 +45,27 @@ _metrics_collector_supported() {
     [ "$(uname -m)" != "aarch64" ]
 }
 
+_needs_nodetool_fix() {
+    # JDK (8 and 11) updates that include the fix for JDK-8278972 cause nodetool
+    # to break due to the more strict URI parsing. This is fixed in Cassandra as
+    # of 3.11.13 and 4.0.4. Management API images based on Cassandra versions
+    # 3.11.12 and 4.0.3 will need to provide a work around
+    [ "${CASSANDRA_VERSION}" == "3.11.12" -o ${CASSANDRA_VERSION} == "4.0.3" ]
+}
+
+_nodetool_fix() {
+    # create a wrapper around nodetool to switch URI parsing to legacy mode
+    if [ -e /opt/cassandra/bin/nodetool ] ; then
+        cp -a /opt/cassandra/bin/nodetool /opt/cassandra/bin/nodetool.orig
+        # create a wrapper
+        cat << EOF > /opt/cassandra/bin/nodetool
+#!/bin/sh
+/opt/cassandra/bin/nodetool.orig -Dcom.sun.jndi.rmiURLParsing=legacy "$@"
+
+EOF
+    fi
```

Comment:
do you need to `chmod ugo+x` this file too?

Reply:
No because I took a shortcut and did a `cp -a` of the original nodetool file. That preserves the file ownership and permissions. Then this just clobbers the contents.